modules/web/web-app/features/billing/plans/PlanSelect.tsx

```diff
@@ -0,0 +1,74 @@
+import { Input, SelectContent, SelectItem, SelectRoot, SelectTrigger } from '@ui/components'
+import { useState } from 'react'
+
+import { useQuery } from '@/lib/connectrpc'
+import { Plan } from '@/rpc/api/plans/v1/models_pb'
+import { getPlanByExternalId, listPlans } from '@/rpc/api/plans/v1/plans-PlansService_connectquery'
+import { ListPlansRequest_SortBy } from '@/rpc/api/plans/v1/plans_pb'
+
+interface Props {
+  value?: Plan['externalId']
+  onChange: (id: Plan['externalId']) => void
+}
+
+export const PlanSelect = ({ value, onChange }: Props) => {
+  const [search, setSearch] = useState('')
+
+  const getPlanQuery = useQuery(
+    getPlanByExternalId,
+    {
+      externalId: value ?? '',
+    },
+    { enabled: Boolean(value) }
+  )
+
+  const onValueChange = (value: string) => {
+    onChange(value)
+    setSearch('')
+  }
+
+  const plan = getPlanQuery.data
+
+  return (
+    <SelectRoot value={value} onValueChange={onValueChange}>
+      <SelectTrigger className="w-[180px]">
+        {plan?.planDetails?.plan?.name ?? 'Choose one'}
+      </SelectTrigger>
+      <SelectContent>
+        <Input
+          key="input"
+          className="mb-2"
+          placeholder="Search.."
+          autoFocus
+          value={search}
+          onChange={event => setSearch(event.target.value)}
+        />
+        <PlanItems search={search} />
+      </SelectContent>
+    </SelectRoot>
+  )
+}
+
+const PlanItems = ({ search }: { search: string }) => {
+  const query = useQuery(listPlans, {
+    pagination: {
+      limit: 20,
+      offset: 0,
+    },
+    productFamilyExternalId: 'default',
```

Comment:
Same here, how are we supposed to provide the family?

Reply:
here if you don't have a product family in the url we don't have it
so we can have a ProductFamilySelect before the PlanSelect, or we fetch all plans for all product families (by making it productFamilyExternalId optional), but you will need to group by product family in the select (as you can have a plan with the same name (ex: Pro) for multiple product families

Reply:
alright, I did the family aware select